From 6008f359c429beed1029e3d62315a11f5c4e792c Mon Sep 17 00:00:00 2001
From: Wei Yang <Wei.Yang@windriver.com>
Date: Tue, 20 Nov 2012 14:40:40 +0800
Subject: [PATCH 041/158] powerpc/prom: remove the illegal reversed memory
 region

When the MEMORY_START is larger than 0 we need to ensure that kernel
does not reserve the range from 0 to MEMORY_START in memblock. Because
the range is out of the available memory.

Signed-off-by: Wei Yang <Wei.Yang@windriver.com>
---
 arch/powerpc/kernel/prom.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/powerpc/kernel/prom.c b/arch/powerpc/kernel/prom.c
index 8b6f7a9..f9c2473 100644
--- a/arch/powerpc/kernel/prom.c
+++ b/arch/powerpc/kernel/prom.c
@@ -679,6 +679,12 @@ void __init early_init_devtree(void *params)
 #endif
 		reserve_crashkernel();
 	early_reserve_mem();
+	/*
+	 * The range of reserved memory region is out of memory
+	 * while MEMORY_START is larger than 0.
+	 * so we should remove it.
+	 */
+	memblock_free(0, MEMORY_START);
 
 	/*
 	 * Ensure that total memory size is page-aligned, because otherwise
-- 
1.8.1.2

